### ISSUE TITLE
Satellite Surface Reconstruction

### ISSUE THREAD

**Issue by SBCV** [Kai-46/VisSatSatelliteStereo#15]
I've written a [library](https://github.com/SBCV/SatelliteSurfaceReconstruction) that allows to combine the results of VisSat with several state-of-the-art meshing / texturing libraries to obtain texture surface reconstructions from satellite images.

Since this is potentially useful for other VisSat users as well, I was wondering if you want to add a reference somewhere. Feel free to close this issue.

**Comment by Kai-46** [comment]
@SBCV  Thanks for the awesome work! And sorry for the ridiculous late reply. I left the satellite image stereo problem for a while, and now am revisiting it and maintaining the newly written repo [SatelliteSfM](https://github.com/Kai-46/SatelliteSfM), in order to make satellite image processing more computer-vision-like. There are a couple of things to do in this roadmap. If you are interested in contributing, we could chat over emails.

**Comment by SBCV** [comment]
@Kai-46 No worries - it is nice to see that you are still working on this topic. Completely abandoning the project would be extremely pity, since it achieves very impressive results.

The reference to [SatelliteSfM](https://github.com/Kai-46/SatelliteSfM) is super helpful, since I've stumbled over it only recently and that more or less only by accident. I don't know, but maybe you should considering to add a reference also on "https://github.com/Kai-46/VisSatSatelliteStereo" and "https://kai-46.github.io/VisSat/" to increase its visibility.

> There are a couple of things to do in this roadmap. If you are interested in contributing, we could chat over emails.

That sounds great but depends on the specific steps in the roadmap. In the long run, I it is reasonable to adapt the [SatelliteSurfaceReconstruction](https://github.com/SBCV/SatelliteSurfaceReconstruction) to the new api of [SatelliteSfM](https://github.com/Kai-46/SatelliteSfM). Is the roadmap somewhere available?

Btw: Have you considered to replace your [adapted colmap](https://github.com/Kai-46/ColmapForVisSat) (which is a full copy of colmap) with a set of git patch files? I have not checked if it is feasible - but if it is, it allows to integrate the latest changes of colmap directly into the colmap adaption. 

Here is a simple example:
https://github.com/Kitware/Danesfield/tree/master/patches
https://github.com/Kitware/Danesfield/blob/19e14936dc149aea610819f1f135b981ffa572f4/Dockerfile#L100

**Comment by SBCV** [comment]
Looks really good!

Regarding the last point:
> release code to run stereo matching on rectified stereo pairs, including both classical and deep ones

Do the classical methods refer to something like [s2p](https://github.com/centreborelli/s2p) or [micmac](https://github.com/micmacIGN/micmac)?

**Comment by Kai-46** [comment]
@SBCV  Yeah, something like that. But I guess I would prefer allow the users to choose their own stereo matching algorithms, e.g., opencv [cv.StereoBM_create](https://docs.opencv.org/3.4/dd/d53/tutorial_py_depthmap.html), or  [https://github.com/Kai-46/sps-stereo](https://github.com/Kai-46/sps-stereo), or any other state-of-the-art stereo matching algorithms? Or, if they want, they could also use off-the-shelf computer vision optical flow algorithms.
